src/lib/components/pages/episode/index.svelte

```diff
@@ -0,0 +1,171 @@
+    import Search from "$icons/search.svelte";
+    import type { SvelteComponentDev } from "svelte/internal";
+    import tippy from "tippy.js";
+
+    $: episode_id = Number($page.params.id);
+
+    const video_player_mapping: {
+        preferences: {
+            [key: string]: {
+                enabled: boolean;
+                text: string;
+            };
+        };
+        options: {
+            [key: string]: {
+                component: typeof SvelteComponentDev;
+                link: string;
+                class: string;
+                text: string;
+            };
+        };
+    } = {
+        preferences: {
+            lights: {
+                enabled: false,
+                text: "Lights"
+            },
+            auto_play: {
+                enabled: true,
+                text: "Auto Play"
+            },
+            auto_next: {
+                enabled: true,
+                text: "Auto Next"
+            },
+            auto_skip_intro: {
+                enabled: true,
+                text: "Auto Skip Intro"
+            }
+        },
+        options: {
+            download: {
+                component: Download,
+                link: "./",
+                class: "w-[1.4vw]",
+                text: "Download"
+            },
+            prev: {
+                component: Next,
+                link: "./",
+                class: "w-[1.4vw] rotate-180",
+                text: "Previous Episode"
+            },
+            next: {
+                component: Next,
+                link: "./",
+                class: "w-[1.4vw]",
+                text: "Next Episode"
+            }
+        }
+    };
+</script>
+
+<episode-container class="block md:py-[2vw] md:pl-[1vw] md:pr-[3.75vw]">
+    <episode-content class="grid grid-cols-12 md:gap-[5vw]">
+        <video-player class="flex flex-col md:col-span-8 md:gap-[0.75vw]">
+            <player class="w-full md:h-[35vw]">
+                <!-- adding a image for now -->
+                <ImageLoader
+                    src="/images/DemonSlayer-episode.webp"
+                    alt="Episode image"
+                    class="w-full object-cover md:h-[35vw] md:rounded-[0.5vw]"
+                />
+            </player>
+            <video-player-options class="flex items-center justify-between">
+                <preferences class="flex items-center md:gap-[1vw]">
+                    <sub-dub class="flex items-center md:gap-[0.75vw]">
+                        <span class="font-semibold uppercase md:text-[1vw]">sub/dub:</span>
+                        <!-- need to make this a dropdown or add tippy js -->
+                        <button class="btn flex items-center bg-surface-400 leading-none md:gap-[0.5vw] md:rounded-[0.35vw] md:px-[0.75vw] md:py-[0.5vw] md:text-[1vw]">
+                            Vidstreaming (sub)
+                            <Chevron class="md:w-[1vw]" />
+                        </button>
```

Comment:
use tippy

Reply:
Yea we need to add dropdown for some things